src/qiboml/backends/pytorch.py

```diff
@@ -64,7 +65,29 @@ def __init__(self):
         self.np.mod = self.np.remainder
         self.np.right_shift = self.np.bitwise_right_shift
         self.np.sign = self.np.sgn
-        self.np.flatnonzero = lambda x: self.np.nonzero(x).flatten()
+        self.np.flatnonzero = lambda x: self.np.nonzero(x).ravel()
+
+        # set the engine of the quantum info operators
+        self.qinfo.ENGINE = self.np
+        self.qinfo.expm = torch.linalg.matrix_exp
+        self.qinfo.ENGINE.Tensor.astype = self.qinfo.ENGINE.Tensor.type
+        self.qinfo.ENGINE.random.normal = lambda loc, scale, size: torch.normal(
+            torch.tensor(loc).repeat(size), torch.tensor(scale).repeat(size)
+        )
+        self.qinfo.ENGINE.random.standard_normal = lambda size: torch.normal(
+            torch.zeros(size), torch.ones(size)
+        )
+        self.qinfo.ENGINE.random.uniform = (
+            lambda low, high, size: (low - high) * torch.rand(size) + high
+        )
+        self.qinfo.ENGINE.random.randint = torch.randint
```

Comment:
This seems to be a cleaner solution for the same problem that was also in the jax backend (with the class Random being defined inside the backend)